meshtastic/tests/test_meshtasticd_tcp_interface_ci.py

```
@@ -0,0 +1,272 @@
+            parsed.username is not None,
+            parsed.password is not None,
+        )
+    )
+
+
+def _parse_host_and_port(host: str) -> tuple[str, int]:
+    """Parse ``HOST[:PORT]`` into a hostname and TCP port."""
+    if not host:
+        raise ValueError(
+            f"Invalid {MESHTASTICD_HOST_ENV_VAR}={host!r}: host component is empty."
+        )
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)